CONTRIBUTING.md

```diff
@@ -1,54 +1,124 @@
 # Design
 
-The system is implemented as an k8s operator using the [operator-sdk](https://github.com/operator-framework/operator-sdk)
-but is deployed as a DaemonSet instead of Deployment with [filtering](https://github.com/operator-framework/operator-sdk/blob/master/doc/user/event-filtering.md) only events for the DaemonSet pod node.
+The system is implemented as an k8s operator using the
+[operator-sdk](https://github.com/operator-framework/operator-sdk) but is
+deployed as a DaemonSet instead of Deployment with
+[filtering](https://github.com/operator-framework/operator-sdk/blob/master/doc/user/event-filtering.md)
+only events for the DaemonSet pod node.
 
-There are two controllers one for [Node](https://godoc.org/k8s.io/api/core/v1#Node)
-and the other for the CRD NodeNeworkState.
 
-## Components
+# Development
 
-- [Node Controller](docs/developer-guide-node.md) - developer guide and design of `Node` handler.
-- [NodeNetworkState Controller](docs/developer-guide-state.md) - developer guide and design of `NodeNetworkState` handler.
-- [NodeNetworkConfigurationPolicy Controller](docs/developer-guide-policy.md) - developer guide and design of `NodeNetworkConfigurationPolicy` handler.
+## Local Kubernetes cluster
 
-# Development
+See [local virtualized cluster guide](docs/deployment-local-cluster.md) to learn
+how to deploy a cluster that can be used to try kubernetes-nmstate, run e2e
+tests and perform debugging.
+
+## Building
+
+```shell
+# If pkg/apis/ has been changed, run generator to update client code
+make gen-k8s gen-openapi
+
```

Comment:
No longer needed now make handler calls it.

Reply:
It is still handy. e.g. when you adjust API, you need to generate first to start coding and get proper help from your IDE